File: modules/wem-webapp/src/main/webapp/admin/common/js/content/page/text/TextDescriptor.ts
module api.content.page.text {

    export class TextDescriptor extends api.content.page.Descriptor {

    }

    export class TextDescriptorBuilder extends api.content.page.DescriptorBuilder {

        public fromJson(json: api.content.page.text.json.TextDescriptorJson): TextDescriptorBuilder {

            this.setKey(api.content.page.DescriptorKey.fromString(json.key))
            this.setName(new api.content.page.DescriptorName(json.name));
            this.setDisplayName(json.displayName);
            this.setConfig(json.config != null ? new api.form.Form(json.config) : null);
            return this;
        }

        public setKey(value: api.content.page.DescriptorKey): TextDescriptorBuilder {
            this.key = value;
            return this;
        }

        public setName(value: api.content.page.DescriptorName): TextDescriptorBuilder {
            this.name = value;
            return this;
        }

        public setDisplayName(value: string): TextDescriptorBuilder {
            this.displayName = value;
            return this;
        }

        public setConfig(value: api.form.Form): TextDescriptorBuilder {
            this.config = value;
            return this;
        }

        public build(): TextDescriptor {
            return new TextDescriptor(this);
        }
    }
}

File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/text/TextDescriptor.java
package com.enonic.wem.api.content.page.text;

import com.enonic.wem.api.content.page.Descriptor;

public class TextDescriptor
    extends Descriptor<TextDescriptorKey>
{
    private TextDescriptor( final Builder builder )
    {
        super( builder );
    }

    public static TextDescriptor.Builder newTextDescriptor()
    {
        return new Builder();
    }

    public static class Builder
        extends BaseDescriptorBuilder<Builder, TextDescriptorKey>
    {
        private Builder()
        {
        }

        public TextDescriptor build()
        {
            return new TextDescriptor( this );
        }
    }
}
